main.py

```diff
@@ -4,16 +4,25 @@
 from typing import Mapping, NamedTuple, Tuple
 from uuid import UUID
 
+import google.auth
 import jwt
 from flask import Request
+from google.cloud import secretmanager
 from requests import Session
 from requests.exceptions import RequestException
 
 from exceptions import InvalidNotifyKeyError, InvalidRequestError
 
-NOTIFY_API_KEY = os.environ["NOTIFY_API_KEY"]
 NOTIFY_BASE_URL = "https://api.notifications.service.gov.uk/v2"
 
+if (NOTIFY_API_KEY := os.getenv("NOTIFY_API_KEY")) is None:
+    _, project_id = google.auth.default()
+    client = secretmanager.SecretManagerServiceClient()
+    access_secret_version_response = client.access_secret_version(
+        name=f"projects/{project_id}/secrets/notify_api_key/versions/latest"
+    )
+    NOTIFY_API_KEY = access_secret_version_response.payload.data.decode("UTF-8")
```

Comment:
I don't think `NOTIFY_API_KEY` should have been capitalised previously as it's not really a constant. Now that we set this value from either the environment or Secret Manager I think we should change the name to `notify_api_key`.